### PR TITLE
adds amazon corretto crypto provider

### PR DESCRIPTION
**Pull request by mritunjaysharma394**
adds amazon corretto crypto provider

Fixes:

Related:

#### For new package PRs only

- [ ] This PR is marked as fixing a pre-existing package request bug
  - [ ] Alternatively, the PR is marked as related to a pre-existing package request bug, such as a dependency
- [x] REQUIRED - The package is available under an OSI-approved or FSF-approved license
- [x] REQUIRED - The version of the package is still receiving security updates

#### For security-related PRs

- [ ] The security fix is recorded in the [advisories](https://github.com/wolfi-dev/advisories) repo

#### For version bump PRs

- [ ] The `epoch` field is reset to 0

#### For PRs that add patches

- [ ] Patch source is documented

### CODE REVIEW

File: amazon-corretto-crypto-provider.yaml [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+  - runs: |
+      mkdir -p ${{targets.destdir}}/usr/share/java/amazon-corretto-crypto-provider
+      mv build/lib/AmazonCorrettoCryptoProvider.jar ${{targets.destdir}}/usr/share/java/amazon-corretto-crypto-provider/AmazonCorrettoCryptoProvider.jar
+
+update:
+  enabled: true
+  github:
+    identifier: corretto/amazon-corretto-crypto-provider
+    use-tag: true
```
Comment (stormqueen1990) [inline amazon-corretto-crypto-provider.yaml:49]: I think we should let it use releases instead.

File: amazon-corretto-crypto-provider.yaml [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+      repository: https://github.com/corretto/amazon-corretto-crypto-provider
+      expected-commit: 296f3821ee8436ea8a311709c0a89e198b19466c
+      tag: ${{package.version}}
+      recurse-submodules: true
+
+  - working-directory: /home/build/aws-lc
+    pipeline:
+      - uses: patch
+        with:
+          patches: /home/build/aws-lc.patch
```
Comment (stormqueen1990) [inline amazon-corretto-crypto-provider.yaml:34]: Perhaps add a comment specifying where this patch came from?

File: amazon-corretto-crypto-provider.yaml (hunk 1 above)
Comment (mritunjaysharma394) [reply amazon-corretto-crypto-provider.yaml]: Thanks @stormqueen1990 , just changed it

File: amazon-corretto-crypto-provider.yaml (hunk 2 above)
Comment (mritunjaysharma394) [reply amazon-corretto-crypto-provider.yaml]: Just done, feel free to review, thanks a lot!

**Review by stormqueen1990 — APPROVED** [review]

**Pull request merged**